src/cascadia/TerminalApp/CascadiaSettingsSerialization.cpp

```diff
@@ -524,6 +543,46 @@ Profile* CascadiaSettings::_FindMatchingProfile(const Json::Value& profileJson)
     return nullptr;
 }
 
+// Method Description:
+// - Finds the "default profile settings" if they exist in the users settings,
+//   and applies them to the existing profiles. The "default profile settings"
+//   are settings that should be applied to every profile a user has, with the
+//   option of being overridden by explicit values in the profile. This should
+//   be called _after_ the defaults have been parsed and dynamic profiles have
+//   been generated, but before the other user profiles have been loaded.
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void CascadiaSettings::_ApplyDefaultsFromUserSettings()
+{
+    // If `profiles` was an object, then look for the `defaults` object
+    // underneath it for the default profile settings.
+    auto defaultSettings{ Json::Value::null };
+
+    if (const auto profiles{ _userSettings[JsonKey(ProfilesKey)] })
+    {
+        if (profiles.isObject())
+        {
+            defaultSettings = profiles[JsonKey(DefaultSettingsKey)];
+        }
+    }
+
+    if (defaultSettings)
+    {
+        _userDefaultProfileSettings = defaultSettings;
+
+        // Remove the `guid` member from the default settings. That'll
+        // hyper-explode, so just don't let them do that.
+        _userDefaultProfileSettings.removeMember({ "guid" });

```

Comment:
Shouldn't we also remove `hidden`?

Reply:
I mean, people could totally just hide _all_ their profiles though, right? Then only unhide profiles if they want them visible (i.e hide _all_ WSL profiles, and only re-enable the cmd profile)

Reply:
yeah i honestly think that's fine